xapian-core/tests/harness/backendmanager.cc

```diff
@@ -123,16 +125,32 @@ BackendManager::get_database(const std::string &dbname,
     string dbleaf = "db__";
     dbleaf += dbname;
     const string& path = get_generated_database_path(dbleaf);
+
+    // checking if a remote backendmanager is being used
+    bool is_remote_bm = false;
+    if (startswith(get_dbtype(), "remote")) {
+	is_remote_bm = true;
+    }
```

Comment:
Checking a boolean return value with `if` only to set a boolean variable based on the result of the test is a bit of an anti-pattern - it's simpler and clearer to just assign the result in such cases:

```
bool is_remote_bm = startswith(get_dbtype(), "remote");
```

Reply:
changed accordingly.